src/ops/kernels/reshape_kernels.cpp

```diff
@@ -0,0 +1,143 @@
+                    size_t num_elements,
+                    hipStream_t stream) {
+  checkCUDA(hipMemcpyAsync(output_ptr,
+                           input_ptr,
+                           num_elements * sizeof(T),
+                           hipMemcpyDeviceToDevice,
+                           stream));
+}
+
+/*static*/
+template <typename T>
+void backward_kernel(T *input_grad_ptr,
+                     const T *output_grad_ptr,
+                     size_t num_elements,
+                     hipStream_t stream) {
+  float alpha = 1.0f;
+  hipLaunchKernelGGL(HIP_KERNEL_NAME(apply_add_with_scale<T>),
+                     GET_BLOCKS(num_elements),
+                     CUDA_NUM_THREADS,
+                     0,
+                     stream,
+                     input_grad_ptr,
+                     output_grad_ptr,
+                     num_elements,
+                     (T)alpha);
+}
+
+template void forward_kernel<float>(float const *input_ptr,
+                                    float *output_ptr,
+                                    size_t num_elements,
+                                    hipStream_t stream);
+template void forward_kernel<double>(double const *input_ptr,
+                                     double *output_ptr,
+                                     size_t num_elements,
+                                     hipStream_t stream);
+template void forward_kernel<int32_t>(int32_t const *input_ptr,
+                                      int32_t *output_ptr,
+                                      size_t num_elements,
+                                      hipStream_t stream);
+template void forward_kernel<int64_t>(int64_t const *input_ptr,
+                                      int64_t *output_ptr,
+                                      size_t num_elements,
+                                      hipStream_t stream);
+
+template void backward_kernel<float>(float *input_grad_ptr,
+                                     float const *output_grad_ptr,
+                                     size_t num_elements,
+                                     hipStream_t stream);
+template void backward_kernel<double>(double *input_grad_ptr,
+                                      double const *output_grad_ptr,
+                                      size_t num_elements,
+                                      hipStream_t stream);
+template void backward_kernel<int32_t>(int32_t *input_grad_ptr,
+                                       int32_t const *output_grad_ptr,
+                                       size_t num_elements,
+                                       hipStream_t stream);
+template void backward_kernel<int64_t>(int64_t *input_grad_ptr,
+                                       int64_t const *output_grad_ptr,
+                                       size_t num_elements,
+                                       hipStream_t stream);
+
+} // namespace Internal
+} // namespace Reshape
+} // namespace Kernels
+}; // namespace FlexFlow
```

Comment:
why does one have a semicolon and the others don't?

Reply:
Either is fine, I'll remove it for consistency.